expr/functions/divideSeries/function.go

```diff
@@ -78,7 +80,7 @@ func (f *divideSeries) Do(ctx context.Context, e parser.Expr, from, until int64,
 		return nil, errors.New("must be called with 2 series or a wildcard that matches exactly 2 series")
 	}
 
-	alignedSeries := helper.AlignSeries(append(numerators, denominator))
+	alignedSeries := helper.AlignSeries(types.CopyMetricDataSlice(append(numerators, denominator)))
```

Comment:
Possible two-time copy (append, and CopyMetricDataSlice). May be one pass alloc, append from nummunators, append from denominator ?

Reply:
I think it should be fine because it's an array of pointers `[]*types.MetricData`, so it's only `CopyMetricDataSlice([]*MetricData)` copying the actual MetricData. I don't see a way of avoiding a potential copy of the pointers when appending. Let me know if you think it's OK or if I misunderstood you.

Reply:
Not problem, this is not blocking. Cost of copying slice of pointers not too high. May be I'll add some benchmarks later and change them, if needed.